dist/api_data.py

```diff
@@ -550,6 +550,10 @@ def __ge__(self, other):
             periodic checkpoints''',
             min='0', max='100000'),
         ]),
+    Config('checkpoint_cleanup_skip_internal', 'true', r'''
+        do not read internal pages when aggregated durable stop timestamp is zero.
+        (Warning: changing this value will likely increase checkpoint time.)
+    ''', type='boolean'),
```

Comment:
I have a few nits and suggestions here:
1. The string name is not that helpful to a general user/reader to know what it means. How about `checkpoint_skip_internal_pages`? The cleanup aspect of checkpoint is not visible to the user.
2. The description is not very helpful to a general user/reader to know whether they'd want to change it. I suggest something like: "if true, skip reading internal pages during checkpoint cleanup for non-timestamped, logged tables. When false, checkpoint will read internal pages to try to clean up obsolete data."
3. In the warning, I suggest "setting this value to false will ...".

Reply:
Reading @agorrod 's general comment about the string name, it could be something like `checkpoint_cleanup=reclaim_space|skip`.

Also, thinking about this more, I think this should be a configuration string set on the `session->checkpoint()` API instead of `wiredtiger_open`. It would still default to skip, but gives the user the opportunity to periodically call checkpoint with it on. That seems more run-time flexible than having to restart the mongod server or at least gives MongoDB a path to an API to run this in the field. In that case it could be called just `cleanup=reclaim_space|skip`.

Reply:
@sueloverso Certainly these changes made; and I would like to nail down the desired interface before implementing for efficiency as much of the code base is still fairly new to me.

My briefing was that this option was for TSE rather than a typical user.

I'd like to confirm if the configuration option is added to checkpoint I'm assuming it will be available during startup, should the user want to _cleanup_ on startup.

@agorrod in earlier comments (and in offline discussion) wondered if it would be better to give it a name that spoke to intent rather than refer to algorithmic details.

If I fuse your suggestions together, would something like: `cleanup=aggressive|relaxed` be appropriate?